### PR TITLE
#10036 Replace codename Merge with Bellatrix (1st step)

### LINKED ISSUE #10036: Replace codename `Merge` with `Bellatrix`
Throughout the Prysm code base, we used `Merge` as the code name for the next hard fork. Finally, as of earlier this week, the community has decided `Bellatrix` to be the next consensus layer hard fork code name. We can now replace all the instances of `Merge` to `Bellatrix` ⭐

### PR DESCRIPTION
**Pull request by leolara**
**What type of PR is this?**

Refactoring

**What does this PR do? Why is it needed?**

Replace codename Merge with Bellatrix 

**Which issues(s) does this PR fix?**

Fixes #10036 

**Other notes for review**

It does not complete 100% the task #10036 another PR will be necessary

### CODE REVIEW

**Comment by leolara** [comment]
It is not passing in my local system the test `TestEndToEnd_MinimalConfig_ValidatorAtCurrentRelease/chain_started` but it does not pass it also without any changes, so I guess I have a local problem, and this is not breaking any tests.

The tests I am running is `bazel test //testing/endtoend:go_default_test` I am not sure if I need to run other tests.

File: beacon-chain/state/stateutil/state_hasher.go [hunk 1]
```diff
@@ -371,15 +371,15 @@ func (h *stateRootHasher) ComputeFieldRootsWithHasherAltair(ctx context.Context,
 }
 
 // ComputeFieldRootsWithHasherMerge hashes the provided merge state and returns its respective field roots.
-func (h *stateRootHasher) ComputeFieldRootsWithHasherMerge(ctx context.Context, state *ethpb.BeaconStateMerge) ([][]byte, error) {
+func (h *stateRootHasher) ComputeFieldRootsWithHasherMerge(ctx context.Context, state *ethpb.BeaconStateBellatrix) ([][]byte, error) {
```
Comment (rkapka) [inline beacon-chain/state/stateutil/state_hasher.go:374]:
```suggestion
func (h *stateRootHasher) ComputeFieldRootsWithHasherBellatrix(ctx context.Context, state *ethpb.BeaconStateBellatrix) ([][]byte, error) {
```

File: beacon-chain/state/stateutil/state_hasher.go [hunk 2]
```diff
@@ -371,15 +371,15 @@ func (h *stateRootHasher) ComputeFieldRootsWithHasherAltair(ctx context.Context,
 }
 
 // ComputeFieldRootsWithHasherMerge hashes the provided merge state and returns its respective field roots.
```
Comment (rkapka) [inline beacon-chain/state/stateutil/state_hasher.go:373]:
```suggestion
// ComputeFieldRootsWithHasherBellatrix hashes the provided merge state and returns its respective field roots.
```

**Review by rkapka — CHANGES_REQUESTED** [review]

**Comment by rkapka** [comment]
> Following files need to be properly formatted:
config/params/mainnet_config.go
proto/prysm/v1alpha1/beacon_state.pb.go

**Comment by rkapka** [comment]
Hey @leolara , you still need to run `gofmt` on a few files:

> Following files need to be properly formatted:
config/params/mainnet_config.go
proto/prysm/v1alpha1/beacon_state.pb.go
runtime/version/fork.go

**Comment by leolara** [comment]
@rkapka thank you for reminding me that. This PR is marked as WIP. The merge of develop into has caused me so git problems with a commit I did at the same time

**Comment by leolara** [comment]
There are lots of things to rename, I will let you know when I think I am finished, so we can review @rkapka . I will try to also rebase on develop after that so it is fresher.

Is there a bazel way of running gofmt? Or I just run it normally?

**Comment by potuz** [comment]
> There are lots of things to rename, I will let you know when I think I am finished, so we can review @rkapka . I will try to also rebase on develop after that so it is fresher.
> 
> Is there a bazel way of running gofmt? Or I just run it normally?

You can run it normally or even as a pre commit hook. I recommend you mark the PR as draft, since then reviewers will get an automated email when you mark it requesting a review

**Comment by leolara** [comment]
@Potuz

> > There are lots of things to rename, I will let you know when I think I am finished, so we can review @rkapka . I will try to also rebase on develop after that so it is fresher.
> > Is there a bazel way of running gofmt? Or I just run it normally?
> 
> You can run it normally or even as a pre commit hook. I recommend you mark the PR as draft, since then reviewers will get an automated email when you mark it requesting a review

Thanks, I changed it to draft

**Comment by leolara** [comment]
I want to note that there is more than a thousand "merge" string instances in the code, and that some times it actually means the action of merge, as opposed to the hard-fork. So I have to read it 🤣

**Comment by terencechain** [comment]
> I want to note that there is more than a thousand "merge" string instances in the code, and that some times it actually means the action of merge, as opposed to the hard-fork. So I have to read it 🤣

If it makes it easier, you don't have to replace everything at once. Open a PR package by package is perfectly fine

File: beacon-chain/core/blocks/payload_test.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/prysmaticlabs/prysm/testing/util"
 )
 
-func Test_MergeComplete(t *testing.T) {
+func Test_BellatrixComplete(t *testing.T) {
```
Comment (terencechain) [inline beacon-chain/core/blocks/payload_test.go:15]: This refers to the action of merge

Comment (leolara) [reply beacon-chain/core/blocks/payload_test.go:15]: Thanks @terencechain , this one slipped :-)

**Comment by leolara** [comment]
> > I want to note that there is more than a thousand "merge" string instances in the code, and that some times it actually means the action of merge, as opposed to the hard-fork. So I have to read it 🤣
> 
> If it makes it easier, you don't have to replace everything at once. Open a PR package by package is perfectly fine

I am doing it more subject by subject, like I am now changing comments and will send that as a commit. If you think the PR is going to get too long to review, then I could do it in several PRs. I think I am kind of half way

**Comment by leolara** [comment]
In order to do not make the review too tedious and complicated, I have decided to stop considering it WIP, get it reviewed and merged, and then I will continue with a follow up PR with the rest of the renaming.

I have rebased on develop, no need to merge develop now.

Please review @terencechain or other :-)

File: config/params/mainnet_config.go [hunk 4]
```diff
@@ -237,6 +237,6 @@ var mainnetBeaconConfig = &BeaconChainConfig{
 	// Light client
 	MinSyncCommitteeParticipants: 1,
 
-	// Merge
+	// Bellatrix
```
Comment (terencechain) [inline config/params/mainnet_config.go:240]: This is a close one, but I think I like `Merge` better here. `TerminalBlockHashActivationEpoch` is used for merging the two chains

File: proto/prysm/v1alpha1/beacon_block.proto [hunk 5]
```diff
@@ -33,7 +33,7 @@ message GenericSignedBeaconBlock {
         // Representing a signed, post-Altair fork beacon block.
         SignedBeaconBlockAltair altair = 2;
 
-        // Representing a signed, post-Merge fork beacon block.
+        // Representing a signed, post-Bellatrix fork beacon block.
         SignedBeaconBlockMerge merge = 3;
```
Comment (terencechain) [inline proto/prysm/v1alpha1/beacon_block.proto:37]:
```suggestion
        SignedBeaconBlockBellatrix bellatrix = 3;
```

Just a reminder here so we don't forget. We can do this in a follow-up PR

**Review by terencechain — DISMISSED** [review]

**Comment by leolara** [comment]
@terencechain I rebased to fix conflict, it seems it needs further action from mantainers to merge

File: config/params/mainnet_config.go (hunk 4 above)
Comment (leolara) [reply config/params/mainnet_config.go:240]: Done

File: beacon-chain/db/kv/schema.go [hunk 6]
```diff
@@ -46,8 +46,8 @@ var (
 
 	// Below keys are used to identify objects are to be fork compatible.
 	// Objects that are only compatible with specific forks should be prefixed with such keys.
-	altairKey = []byte("altair")
-	mergeKey  = []byte("merge")
+	altairKey    = []byte("altair")
+	bellatrixKey = []byte("bellatrix")
```
Comment (prestonvanloon) [inline beacon-chain/db/kv/schema.go:50]: This is going to break everyone on testnets. Do not rename database keys without a migration.

File: config/params/mainnet_config.go (hunk 4 above)
Comment (prestonvanloon) [reply config/params/mainnet_config.go:240]: @leolara it still says Bellatrix. Did you intend to change it back to "Merge" per Terence's comment?

File: proto/prysm/v1alpha1/beacon_state.pb.go [hunk 7]
```diff
@@ -1073,7 +1073,7 @@ func (x *SyncAggregatorSelectionData) GetSubcommitteeIndex() uint64 {
 	return 0
 }
 
-type BeaconStateMerge struct {
+type BeaconStateBellatrix struct {
```
Comment (prestonvanloon) [inline proto/prysm/v1alpha1/beacon_state.pb.go:1076]: Renaming in the generated code is not ideal. Please re-run `./hack/update-go-pbs.sh` after modifying proto files.

The reason is that `file_proto_prysm_v1alpha1_beacon_state_proto_rawDesc` attribute should be updated as well and it's some hex bytes so rebuilding the pb.go files is really the only way to do it. You may need to run gofmt or goimports to sort the imports for the linter.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: proto/prysm/v1alpha1/beacon_state.pb.go (hunk 7 above)
Comment (leolara) [reply proto/prysm/v1alpha1/beacon_state.pb.go:1076]: ok

File: beacon-chain/db/kv/schema.go (hunk 6 above)
Comment (leolara) [reply beacon-chain/db/kv/schema.go]: Reverted

File: proto/prysm/v1alpha1/beacon_state.pb.go (hunk 7 above)
Comment (leolara) [reply proto/prysm/v1alpha1/beacon_state.pb.go:1075]: Done, many files regenerated and pushed

File: config/params/mainnet_config.go (hunk 4 above)
Comment (leolara) [reply config/params/mainnet_config.go]: Done

**Comment by leolara** [comment]
@prestonvanloon @terencechain @rkapka I think all comments done, rebased to resolve conflicts.

Do you want me to squash when merging or merge normally?

Please, I think we need to move this, as this touches so many files many PRs will conflict this one

**Comment by leolara** [comment]
It seems the problems are not only conflicts with new additions, but also the use of symbols I have renamed in other PRs.

For example, I renamed version.Merge to version.Bellatrix, if someone uses version.Merge and merges a PR, then that symbol does not exist in my PR.

I think we should freeze merging other PRs until this is merged or it will be very difficult

**Comment by prestonvanloon** [comment]
@leolara You need to run gofmt and goimports after regenerating protofiles.

```
gofmt -w **/*.go
goimports -w **/*.go
```

Git/Github is not letting me push to your branch. Please fix and we can merge quickly.

**Comment by leolara** [comment]
@prestonvanloon I ran those commands, but there were no effect, it seems my branch is clean or the problem is other

**Comment by leolara** [comment]
@prestonvanloon I changed a setting try to push to my branch now

**Review by prestonvanloon — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**